cpp/src/parquet/thrift_internal.h

```diff
@@ -363,12 +364,26 @@ static inline format::EncryptionAlgorithm ToThrift(EncryptionAlgorithm encryptio
 
 using ThriftBuffer = apache::thrift::transport::TMemoryBuffer;
 
+// On Thrift 0.14.0+, we want to use TConfiguration to raise the max message size
+// limit (ARROW-13655).  If we wanted to protect against huge messages, we could
+// do it ourselves since we know the message size up front.
+
+inline std::shared_ptr<ThriftBuffer> CreateReadOnlyMemoryBuffer(uint8_t* buf,
+                                                                uint32_t len) {
+#if PARQUET_THRIFT_VERSION_MAJOR > 0 || PARQUET_THRIFT_VERSION_MINOR >= 14
+  auto conf = std::make_shared<apache::thrift::TConfiguration>();
+  conf->setMaxMessageSize(std::numeric_limits<int>::max());
+  return std::make_shared<ThriftBuffer>(buf, len, ThriftBuffer::OBSERVE, conf);
```

Comment:
[The `ThriftBuffer` (aka `TMemoryBuffer`) constructor sets the max limit to the buffer as `std::numeric_limits<uint32_t>::max()`](https://github.com/apache/thrift/blob/master/lib/cpp/src/thrift/transport/TBufferTransports.h#L463). Is the intent here to use `int` as to cap at a lower limit?

Reply:
This is different.We're talking about `maxMessageSize`, not `maxBufferSize`.

Reply:
I seem to recall in the past thrift wasn't good about memory bombs produced by the fuzzer is this fixed in 14?

Reply:
See a bit below. `setStringSizeLimit` and `setContainerSizeLimit` should protect us against Thrift memory bombs, IIRC. However, we don't have any protection against compression memory bombs.